src/core/server/logging/logging_config.ts

```diff
@@ -93,6 +93,23 @@ export const config = {
 
 export type LoggingConfigType = TypeOf<typeof config.schema>;
 
+export const loggerContextConfigSchema = schema.object({
+  appenders: schema.mapOf(schema.string(), Appenders.configSchema, {
+    defaultValue: new Map<string, AppenderConfigType>(),
+  }),
+
+  loggers: schema.arrayOf(createLoggerSchema, { defaultValue: [] }),
```

Comment:
> Another question I had: I know it was already the case before this PR, but I'm not sure to see why we used an array for loggers instead of a context->config mapping, as we can't define multiple loggers with the same context anyway.

@azasypkin knows better. I suspect it has been done to follow log4j format.
Oleg, you also mentioned that it could create problems in a system when runtime has several different configuration files (ECC, for example) and it needs to merge them correctly.

Reply:
> @azasypkin knows better. I suspect it has been done to follow log4j format.
> Oleg, you also mentioned that it could create problems in a system when runtime has several different configuration files (Elastic Cloud, for example) and it needs to merge them correctly.

That's correct, I was inspired by log4j format and as I said the other day, I think it was a bad decision :slightly_smiling_face:  E.g. in Cloud (or probably in a Docker env too) you won't be able to add new, remove or update (enable/disable/change level) existing logger if we continue using an array format. In Cloud you may want to temporarily upgrade logging level for some specific context without affecting user overrides, but with array format it's all or nothing. We cannot just do something like this (`security.session` is context):
```yaml
loggers.security.session.level: 'debug'
```

Reply:
I'm fine to switch to `Map`, providing that elasticsearch allows users to use this syntax
https://github.com/elastic/elasticsearch/blob/14de76e53f833983328c670c881cb58e8aafffb7/x-pack/plugin/core/src/main/config/log4j2.properties#L75

Reply:
Created a follow-up issue: https://github.com/elastic/kibana/issues/69592

I'll address this immediately following this PR.